storage/aptosdb/src/lib.rs

```diff
@@ -926,16 +964,25 @@ impl DbReader for AptosDB {
         })
     }
 
-    /// Get write sets for range [begin_version, end_version).
+    /// Returns write sets for range [begin_version, end_version).
     ///
     /// Used by the executor to build in memory state after a state checkpoint.
-    /// Any missing write set in the entire range results in error.
+    /// Any missing write set in the entire range results in an error.
     fn get_write_sets(
         &self,
         begin_version: Version,
         end_version: Version,
     ) -> Result<Vec<WriteSet>> {
         gauged_api("get_write_sets", || {
+            if let Some(pruner) = self.pruner.as_ref() {
```

Comment:
Can you create a helper function for this check and reuse that across the APIs to reduce duplication?

Reply:
Done